fsharp-backend/src/LibExecutionStdLib/LibString.fs

```diff
@@ -533,20 +533,17 @@ let fns : List<BuiltInFn> =
       parameters = [ Param.make "s" TStr ""; Param.make "separator" TStr "" ]
       returnType = TList TStr
       description =
-        "Splits a string at the separator, returning a list of strings without the separator. If the separator is not present, returns a list containing only the initial string."
+        "Splits a string at the separator, returning a list of strings without the separator.
+        If the separator is not present, returns a list containing only the initial string."
       fn =
         (function
         | _, [ DStr s; DStr sep ] ->
+          // This behaviour is the worst - it mimics what OCaml did:
+          // There are (n-1) empty strings returned for each sequence of n strings
+          // matching the separator (eg: split "aaaa" "a" = ["", "", ""]).
           if sep = "" then
             s |> String.toEgcSeq |> Seq.toList |> List.map DStr |> DList |> Ply
           else
-            // CLEANUP: we need a new version of this fn.
-            // This behaviour is the worst. This mimics what OCaml did: There
-            // should be (n-1) empty strings returned for each sequence of n
-            // strings matching the separator (eg: split "aaaa" "a" = ["",
-            // "", ""]). However, the .NET string split puts in n-1 empty
-            // strings correctly everywhere except at the start and end,
-            // where there are n empty strings instead.
```

Comment:
I wasn't sure what this bit meant - are you happy with .NET's String.split, or no?

Reply:
This is one of those things where there isn't necessarily a right answer. Different languages and libraries take different approaches, and there's good reasons for different choices. Our old version didn't have good reasons for its choices, and was really quite bad.

So let me turn this around: what should be the behaviour? Are you happy with the test output here? If so, why? If not, what do you think it should do differently?

Reply:
The new implementation here fits what most languages seem to do, which is the best option without having some Dark-specific reason to do things differently, so I'm happy with the test output here for that reason.

Beyond that, I don't have any particular investment here - even the current impl. seems reasonable to me.